pkg/downloader/chart_downloader.go

```diff
@@ -125,6 +137,34 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+// https://github.com/kubernetes/helm/issues/2937
+// tries to fix URL because of BUG #2937
+// transforms URL from
+// https://url.to.repo/path?key=value/chart-0.1.0.tgz
+// into
+// https://url.to.repo/path/chart-0-1-0.tgz?key=value
+func fixupURL(chartURL string) (string, error) {
```

Comment:
This function makes a fair number of assumptions and takes a bit to understand what its doing. If there isn't another way to solve this problem this function needs to have good tests. I would hope there is an easier way to solve this.

Reply:
Thank you. I have removed it and solved the problem at the source. I am not sure, though, if there is any other place that will be impacted.